### PR TITLE
Modularise media state

### LINKED ISSUE #42456: Modularise `media`
The `media` portion of the state tree needs to be modularised. See the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) for more details.

### PR DESCRIPTION
**Pull request by sgomes**
This PR is one of many working on modularising state in Calypso. This one handles media state.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

Fixes #42456.

#### Changes proposed in this Pull Request

* Modularise media state

#### Testing instructions

Unfortunately, media state is loaded pretty early in Calypso boot, due to things like site icons depending on it. As such, it's not straightforward to verify the before/after on the automatic loading process, only that it does load.

Some simple smoke-test of media functionality should be enough to validate that it continues working correctly after this change, as there are no changes to the state functionality itself, only its loading.

### CODE REVIEW

**Comment by sgomes** [comment]
Please note that @matticbot still calculates totals incorrectly, so things are nowhere near as bad as the above makes it seem. The entire application only grows by a net 6KB compressed, and most users are likely to get a net positive, rather than a net negative, from this change.

See [the second table on this page](http://iscalypsofastyet.com/p/branch?branch=update/modularise-media-state) for the full details of what happens to the chunks.

File: client/state/selectors/get-media-item-errors.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * Internal dependencies
+ */
+import 'calypso/state/media/init';
```
Comment (tyxla) [inline client/state/selectors/get-media-item-errors.js:4]: Seems like this selector is completely unused. But it should be fine to remove it in another PR.

File: client/state/selectors/is-requesting-media-item.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
 /**
  * External dependencies
  */
-
 import { get } from 'lodash';
 
+/**
+ * Internal dependencies
+ */
+import 'calypso/state/media/init';
```
Comment (tyxla) [inline client/state/selectors/is-requesting-media-item.js:9]: Seems like this selector is completely unused. But it should be fine to remove it in another PR.

File: client/state/selectors/is-fetching-next-page.js [hunk 3]
```diff
@@ -2,8 +2,14 @@
  * External dependencies
  */
 import { get } from 'lodash';
+
```
Comment (tyxla) [inline client/state/selectors/is-fetching-next-page.js:5]: Side note: this selector's name is way too broad 😬

**Review by tyxla — APPROVED** [review]
Looks good and tests well. Nice 👏  

🚢  it!

File: client/state/selectors/is-fetching-next-page.js (hunk 3 above)
Comment (sgomes) [reply client/state/selectors/is-fetching-next-page.js:5]: One parallel effort we should be thinking about is reorganising state.

I seem to recall we had decided that selectors for a particular area of state should live under `state/{area}/selectors`, and only selectors that draw from multiple areas of state should be in `state/selectors`.

In that situation, `state/media/is-fetching-next-page` would be perfectly fine.

**Comment by sgomes** [comment]
Thank you for another thorough review, @tyxla! 🙇

File: client/state/selectors/is-fetching-next-page.js (hunk 3 above)
Comment (tyxla) [reply client/state/selectors/is-fetching-next-page.js:5]: Agreed. 

I haven't checked thoroughly, but many of the selectors in the so-called "global selectors directory" were put there because it was considered it was going to be the place for ALL selectors at some point in history. As you pointed out, that has changed since, but we haven't really done a thorough audit to move away some of them that are specific to a certain state tree.
